src/rgw/rgw_auth.h

```diff
@@ -450,11 +473,12 @@ class LocalApplier : public IdentityApplier {
                const RGWUserInfo& user_info,
                std::string subuser)
     : user_info(user_info) {
-    parse_subuser_info(subuser, user_info);
+    std::pair<const std::string&, uint32_t> subuser_info = parse_subuser_info(subuser, user_info);
+    subuser = subuser_info.first;
```

Comment:
The concept was a bit different. Having the `parse_subuser_info` producing output via the return value allows to use solely initialisation list, and thus preserve const-correctness for the whole class. The value it produces (the `std::pair` or, even more preferably, the `struct` mentioned earlier) can be stored as `const` and then only appropriately *extracted* in `get_perm_mask` and `get_subuser_name`.